runtime/src/bpf_map/userspace/prog_array.cpp

```diff
@@ -11,34 +11,20 @@
 #include <cerrno>
 #include <spdlog/spdlog.h>
 #include <bpf/libbpf.h>
-
-// Why hand written syscalls? syscall() function was hooked by syscall server,
-// direct call to it will lead to a result provided by bpftime. So if we want to
-// get things from kernel, we must manually execute `syscall`
+#include <dlfcn.h>
+#include <gnu/lib-names.h>
 
 #ifndef offsetofend
 #define offsetofend(TYPE, FIELD)                                               \
 	(offsetof(TYPE, FIELD) + sizeof(((TYPE *)0)->FIELD))
 #endif
-__attribute__((__noinline__, optnone, noinline)) static long
-my_bpf_syscall(long cmd, union bpf_attr *attr, unsigned long size)
-{
-	long ret;
-	__asm__ volatile("movq %1, %%rax\n"
-			 "movq %2, %%rdi\n"
-			 "movq %3, %%rsi\n"
-			 "movq %4, %%rdx\n"
-			 "movq $0, %%r10\n"
-			 "movq $0, %%r8\n"
-			 "movq $0, %%r9\n"
-			 "syscall\n"
-			 "movq %%rax, %0"
-			 : "=g"(ret)
-			 : "i"((long)__NR_bpf), "g"(cmd), "g"(attr), "g"(size)
-			 : "memory", "rdi", "rsi", "rdx", "r10", "r8", "r9",
-			   "rax");
-	return ret;
-}
+
+// syscall() function was hooked by syscall server, direct call to it will lead to 
+// a result provided by bpftime. So if we want to get things from kernel, we must 
+// manually execute `syscall` from libc
+static void* libc_handle = dlopen(LIBC_SO, RTLD_LAZY);
+static auto libc_syscall = reinterpret_cast<decltype(&::syscall)>(
```

Comment:
We need pay attention to all global variable initializations, since attaching might be executed before the initialization of global variables